src/views/PositionPdfPlot/PositionPdfPlotWidget.tsx

```diff
@@ -70,103 +71,98 @@ const PositionPdfPlotWidget: FunctionComponent<Props> = ({fetchSegment, startTim
     useRecordingSelectionTimeInitialization(startTimeSec, endTimeSec)
     const { visibleTimeStartSeconds, visibleTimeEndSeconds } = useTimeRange()
     const numTimepoints = Math.floor((endTimeSec - startTimeSec) * samplingFrequency)
-    const dataModel = usePositionPdfDataModel(fetchSegment, numTimepoints, numPositions, segmentSize, multiscaleFactor)
+    const dataModel = usePositionPdfDataModel(fetchSegment, numPositions, segmentSize)
     const [showLinearPositionsOverlay, setShowLinearPositionsOverlay] = useState<boolean>(false)
+    
+    const rangeStartSample = useMemo(() => {
+        return visibleTimeStartSeconds === undefined ? 0 : Math.max(0, Math.floor(visibleTimeStartSeconds - startTimeSec) * samplingFrequency)
+    }, [visibleTimeStartSeconds, startTimeSec, samplingFrequency])
+    const rangeEndSample = useMemo(() => {
+        return visibleTimeEndSeconds === undefined ? 0 : Math.min(numTimepoints, Math.ceil((visibleTimeEndSeconds - startTimeSec) * samplingFrequency))
+    }, [visibleTimeEndSeconds, numTimepoints, startTimeSec, samplingFrequency])
```

Comment:
I broke some of these computations out into individual memoized quantities--I don't fully trust destructuring of return results in the context of memoization, but more importantly I hope this makes it a bit more straightforward what the quantities in question refer to.